### ISSUE TITLE
False negatives on SA_FIELD_SELF_COMPUTATION and SA_LOCAL_SELF_COMPUTATION

### ISSUE THREAD

**Issue by gloNelson** [spotbugs/spotbugs#1473]
There are potential false negatives on **SA_FIELD_SELF_COMPUTATION** and **SA_LOCAL_SELF_COMPUTATION.**
I found that their methods support `int`, `short`, `long`, but not `double`, `float`, or any other types according to [FindSelfComparison.java](https://github.com/spotbugs/spotbugs/blob/a6f9acb2932b54f5b70ea8bc206afb552321a222/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindSelfComparison.java#L268-L273).

``` java
// #L268 - #L273
if ("D".equals(item0.getSignature()) || "F".equals(item0.getSignature())) {
    return;
}
if ("D".equals(item1.getSignature()) || "F".equals(item1.getSignature())) {
    return;
}
```
It can be seen from the code that it will skip check for self-operation when one of two opcodes is `double` or `float` type.
In that case, there will be potential false negatives. 
e.g
```java
double testSA_LOCAL_SELF_COMPUTATION(){
      double flags = 1;
      return flags - flags; // no warning of SA_LOCAL_SELF_COMPUTATION (False negative)
}

double x = 1;
double testSA_FIELD_SELF_COMPUTATION2(){
    return x - x; // no warning of SA_FIELD_SELF_COMPUTATION (False negative)
}
```
So even other Java class types.
```java
double temp = new java.util.Date().getTime();
double testSA_FIELD_SELF_COMPUTATION4(){
    double second = temp - temp; // no warning of SA_FIELD_SELF_COMPUTATION (False negative)
    return second;
}

double testSA_FIELD_SELF_COMPUTATION4(java.util.Date endDate){
    double temp1 = endDate.getTime();
    double second = temp1 - temp1; // no warning of SA_LOCAL_SELF_COMPUTATION (False negative)
    return second;
}
```

**Comment by gloNelson** [comment]
Another problematic code section in FindSelfComparison.java is mentioned in #1472 (SA_LOCAL_SELF_COMPUTATION bug )

**Comment by gloNelson** [comment]
Same problem in `SA_FIELD_SELF_COMPARISON` and `SA_LOCAL_SELF_COMPARISON `
